### PR TITLE
feat: working repo submit button

### LINKED ISSUE #352: Connect the Submit a Repository button to the API (POST)
## What is needed?
The submit button in the dropdown sends a link to the discord when submitted. We need also to connect it to the submit service in the API. This will require a new [hook](https://github.com/open-sauced/hot/tree/alpha/src/hooks) (API hooks only exist on the alpha branch at the moment) called **useRepoSubmit**.

Screen Shot 2022-10-03 at 11 00 16 PM

## Things to consider
Submit the repo using the id - https://api.opensauced.pizza/#/Submit%20service/submitOneById

Screen Shot 2022-10-03 at 10 59 47 PM

### PR DESCRIPTION
**Pull request by bdougie**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

repo submission handler function is added, and tried adding existing repos like freeCodeCamp or opensauced hot repo, but it's getting me back 500 or 404 response.

1. Sign in.
2. Click User Emoticon at top right corner, and click submit repo.
3. Type in repo URL (I tried putting freeCodeCamp's repo & OpenSauced/hot repo)
4. API call returns an error with 500 or 404 response. 

## Related Tickets & Documents

Resolves #352 

## Mobile & Desktop Screenshots/Recordings

Screen Shot 2022-11-03 at 2 26 36 PM
Screen Shot 2022-11-03 at 2 26 57 PM

## updated approach
I took this over #394, and solved the problem of handling error codes. I will work on #404 next to start queueing repos that are not indexed in api.opensauced.pizza. 

Screen Shot 2022-11-29 at 4 34 47 PM

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
Work on the #404 and this repo - https://github.com/open-sauced/open-sauced-goals - to make a queue for indexing.

### CODE REVIEW

File: src/components/RepoSubmission.tsx [hunk 1]
```diff
@@ -38,11 +40,42 @@ const RepoSubmission = ({ isFormOpen, handleFormOpen }: RepoSubmissionProps): JS
...
+        const resp = await submitRepo(sanitizedUrl, currentUser.access_token);
+
+        try {
+          if (resp.status === 200) {
+            ToastTrigger({ message: "Repo submitted successfully", type: "success" });
+            // issue #409 - TODO: This sendMessage is not working
+            sendMessage(sanitizedUrl, userName);
+          }
+
+          if (resp.status === 404) {
```
Comment (bdougie) [inline src/components/RepoSubmission.tsx:54]: Hey, @janie-lee-developer per our conversation last week. This approach is what was needed to push this over the finish line. I will now issue #404 in a separate PR. Let me know if you have questions.

**Comment by bdougie** [comment]
@open-sauced/triage this one is ready to QA.

File: src/components/RepoSubmission.tsx [hunk 2]
```diff
@@ -12,6 +13,7 @@ export declare interface RepoSubmissionProps {
 
 const RepoSubmission = ({ isFormOpen, handleFormOpen }: RepoSubmissionProps): JSX.Element => {
   const { user } = useSupabaseAuth();
```
Comment (0-vortex) [inline src/components/RepoSubmission.tsx:15]: this is `currentUser.user`, the hook could be manipulated to directly export `currentUser.access_token` and deduplicate some code below

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Very minor code change to reduce confusion next time we need the token 🙏

File: src/components/RepoSubmission.tsx (hunk 2 above)
Comment (bdougie) [reply src/components/RepoSubmission.tsx]: Just push an update to export token from the hook.

File: src/components/RepoSubmission.tsx (hunk 1 above)
Comment (janie-lee-developer) [reply src/components/RepoSubmission.tsx:52]: Got it! Thank you for tagging me!!

**Review by 0-vortex — APPROVED** [review]
LGTM! 👍

**Pull request merged**